### ISSUE TITLE
Values stay as "override" even when repo comes in line

### ISSUE THREAD

**Issue by abangser** [argoproj/argo-cd#5076]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

It is possible to override a parameter by changing the value in the UI. One use case for this is to set a value in the UI to validate it can roll out OK before committing the value to code. If your parameter override becomes equivalent to the value in code, there is still a parameter override pill on the Application and warning sign on the field value, but there is no way to "remove" the override.

**To Reproduce**

Pre-req: Confirm auto-sync is turned off
1. Commit code with a parameter set at the application level (e.g. `Foo: bar`)
2. Override a parameter value in the UI at the application level (e.g. `Foo: baz`)
3. Confirm there is an `override` marker on the Application and a ⚠️ marker on the property itself
4. Allow the new parameter to be rolled out
5. Commit code with the new parameter value set (e.g. `Foo: baz`)
6. Confirm the app syncs this new code
7. UNEXPECTED: The parameter is still overridden with both markers
8. BROKEN: When editing the parameter, it is impossible to remove the override either by link OR by resaving the parameters

**Expected behavior**

Ideally, the two markers for overridden parameters would be removed, the value would be set to the code value `Foo: baz`).

But, at a minimum we should have access to the `remove override` link on the right side of the parameter field.

**Screenshots**

Missing link:
![image](https://user-images.githubusercontent.com/1557346/102406094-fd9c4300-3fe1-11eb-9672-833947345417.png)

**Version**

```shell
~ ➜ argocd version                     
argocd: v1.7.3+b4c79cc
  BuildDate: 2020-09-01T23:20:15Z
  GitCommit: b4c79ccb88173604c3786dcd34e83a9d7e8919a5
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.7.7+33c93ae
  BuildDate: 2020-09-29T04:56:23Z
  GitCommit: 33c93aea0b9ee3d02fb9703cd82cecce3540e954
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.17.8
```

**Comment by alexmt** [comment]
I agree, this is valid bug, I've also noticed it. Issue has a workaround - parameter still can be removed in the UI using  Spec tab.

**Comment by abangser** [comment]
Ah interesting work around! I just changed it to something else which then gave me the Remove Override link available again. This was safe in my circumstances but that other workaround would be safer overall I think.

**Comment by dannysauer** [comment]
Deleting the overridden parameter name _and_ value from the manifest tab resolved the issue for me as well (I assume Spec/Manifestis a localization diff).

**Comment by drewboswell** [comment]
this is still a problem. It is doubly disturbing as these overrides don't pop as the app being out of sync.

**Comment by ClaytonJY** [comment]
This bug still exists, but so does the workaround described above, though "Spec" is now "Manifest".
